cartography/intel/aws/route53.py

```diff
@@ -396,9 +396,11 @@ def link_sub_zones(
     """
     query = """
     MATCH (:AWSAccount{id: $AWS_ID})-[:RESOURCE]->(z:AWSDNSZone)
-        <-[:MEMBER_OF_DNS_ZONE]-(record:DNSRecord{type:"NS"})
-        -[:DNS_POINTS_TO]->(ns:NameServer)<-[:NAMESERVER]-(z2:AWSDNSZone)
-    WHERE record.name=z2.name AND NOT z=z2
+    <-[:MEMBER_OF_DNS_ZONE]-(record:DNSRecord{type:"NS"})
+    -[:DNS_POINTS_TO]->(ns:NameServer)<-[:NAMESERVER]-(z2:AWSDNSZone)
+    WHERE record.name = z2.name AND
+    z2.name ENDS WITH '.' + z.name AND
```

Comment:
I think this is the fix, thank you very much. Previously the query was just connecting zone1 and zone2 if they were sharing the same name servers.